### PR TITLE
Remove unnecessary constraints from collaborator management

### PR DESCRIPTION
**Pull request by kschiffer**
#### Summary
References #939.
This PR removes some constraints from the collaborator form in the console. These constraints have been wrongfully added by me, due to a lack of understanding of how our right intersection works. This has been cleared up in #939.

#### Changes
- Remove a constraint that forbids right modification of users with universal rights (and wrongfully referring to these rights as `admin rights`)
- Modify the logic for correctly handling collaborator rights that are out of the scope of the current user
  - Disable checkboxes for out-of-scope rights
  - Allow modification of remaining rights
  - Fix out-of-scope rights disappearing when unchecked
  - Disallow right modification for users with universal rights (while not having this right oneself), but still allow removal as collaborator, see #1364 
- While at it, fixing prop type warnings of the respective components (#1086) 

#### Notes for Reviewers
This PR will for now only remove unnecessary restrictions when modifying right, without changing the UI logic. I propose to continue on improving the UX here as discussed in https://github.com/TheThingsNetwork/lorawan-stack/issues/1204#issuecomment-528854149 in a separate PR.

#### Release Notes
- Removed unnecessary constraints for the right management of collaborators

cc @htdvisser

### CODE REVIEW

**Comment by htdvisser** [comment]
I don't think we should use the term "scope" here. Instead of _"This entity possesses rights that are out of your scope of granted rights."_, can we do something like _"This {account_type} has rights that you don't have"_?

File: pkg/webui/components/checkbox/checkbox.js [hunk 1]
```diff
@@ -131,36 +137,37 @@ class Checkbox extends React.PureComponent {
...
-  label: PropTypes.message,
-  name: PropTypes.string.isRequired,
-  value: PropTypes.oneOfType([PropTypes.bool, PropTypes.object]),
-  checked: PropTypes.bool,
   disabled: PropTypes.bool,
-  readOnly: PropTypes.bool,
-  autoFocus: PropTypes.bool,
-  onChange: PropTypes.func,
+  indeterminate: PropTypes.bool,
+  label: PropTypes.oneOfType([PropTypes.message, PropTypes.node]),
```
Comment (bafonins) [inline pkg/webui/components/checkbox/checkbox.js]: use `children` for custom content

File: pkg/webui/console/components/rights-group/index.js [hunk 2]
```diff
@@ -140,15 +146,21 @@ class RightsGroup extends React.Component {
         className={style.rightLabel}
         key={right}
         name={right}
-        label={{ id: `enum:${right}` }}
         disabled={outOfOwnScopeRights.includes(right)}
+        label={
+          <React.Fragment>
+            <Message content={{ id: `enum:${right}` }} />{' '}
+            {outOfOwnScopeRights.includes(right) && (
+              <Message className={style.revoke} content={m.revocableOnly} />
```
Comment (bafonins) [inline pkg/webui/console/components/rights-group/index.js]: I would prefer making this a custom checkbox component

File: pkg/webui/console/components/rights-group/index.js [hunk 3]
```diff
@@ -50,41 +52,45 @@ const computeState = function(values, rights, universalRight) {
...
+    const { value, rights: grantableRights, universalRight: grantableUniversalRight } = props
+    const { outOfOwnScopeRights } = state
+    let universalRight = grantableUniversalRight
+    const allGrantableRights = [...grantableRights, ...grantableUniversalRight]
+    let derivedOutOfOwnScopeRights = outOfOwnScopeRights
+
+    // Initially identify given values that are out of the scope of the
+    // current user, to be able to persist them in the rights list
+    if (outOfOwnScopeRights.length === 0) {
+      derivedOutOfOwnScopeRights = value.filter(function(right) {
```
Comment (bafonins) [inline pkg/webui/console/components/rights-group/index.js]: this is hard to read. could you try refactoring it a bit?

File: pkg/webui/console/components/rights-group/index.js (hunk 2 above)
Comment (kschiffer) [reply pkg/webui/console/components/rights-group/index.js]: I think to pass the additional content as children will simplify this enough.

**Comment by kschiffer** [comment]
> The 'check all' checkbox label changes when uncheking and checking again:
> Screenshot 2019-08-22 at 08 48 24

How did you get to this state?

> Also, is it expected behaviour that the user with several rights can update rights of the creator of the app (having all possible rights and being an admin), but cannot change rights of the user that has just more rights?

See #1204. This will be changed as soon as the issue is resolved.

**Comment by bafonins** [comment]
> How did you get to this state?

Create a new user with several rights and try to edit the user with all rights

**Comment by bafonins** [comment]
blocked by https://github.com/TheThingsNetwork/lorawan-stack/pull/1345

**Comment by kschiffer** [comment]
Updated OP. Changes resulting from #1345 are now integrated.
> This PR will for now only remove unnecessary restrictions when modifying right, without changing the UI mechanic. I propose to continue on improving the UX here as discussed in https://github.com/TheThingsNetwork/lorawan-stack/issues/1204#issuecomment-528854149 in a separate PR.

**Pull request closed without merging**